src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BindLoweringPass.cs

```diff
@@ -190,23 +191,32 @@ private void RewriteUsage(TagHelperIntermediateNode node, int index, ComponentAt
             //
             // BindMethods.GetValue(<code>) OR
             // BindMethods.GetValue(<code>, <format>)
-            //
-            // For now, the way this is done isn't debuggable. But since the expression
-            // passed here must be an LValue, it's probably not important.
-            var valueNodeContent = format == null ?
-                $"{BlazorApi.BindMethods.GetValue}({originalContent})" :
-                $"{BlazorApi.BindMethods.GetValue}({originalContent}, {format})";
             valueAttributeNode.Children.Clear();
-            valueAttributeNode.Children.Add(new CSharpExpressionIntermediateNode()
+
+            var expression = new CSharpExpressionIntermediateNode();
+            valueAttributeNode.Children.Add(expression);
+
+            expression.Children.Add(new IntermediateToken()
             {
-                Children =
+                Content = $"{BlazorApi.BindMethods.GetValue}(",
+                Kind = TokenKind.CSharp
+            });
+            expression.Children.Add(original);
+
+            if (!string.IsNullOrEmpty(format?.Content))
+            {
+                expression.Children.Add(new IntermediateToken()
                 {
-                    new IntermediateToken()
-                    {
-                        Content = valueNodeContent,
-                        Kind = TokenKind.CSharp
-                    },
-                },
+                    Content = ", ",
+                    Kind = TokenKind.CSharp,
+                });
+                expression.Children.Add(format);
+            }
+
+            expression.Children.Add(new IntermediateToken()
+            {
+                Content = ")",
+                Kind = TokenKind.CSharp,
```

Comment:
I had to change how this worked because I realized that we broke completion.

What would happen is that you'd get black text inside the attribute, no colorization, no completion. This is a more involved approach that I was hoping to avoid (and has knock on effects elsewhere, but it's necessary).

Reply:
The root cause is that if you take C# code that the user wrote, and don't map it to a location in the output document, then they won't get C# editor features for that location.

I have an idea how we could write tests for this and I'm going to try it out in Razor.